scripts/irods/test/test_prep_genquery_iterator.py

```diff
@@ -127,7 +133,195 @@ def setUp(self):
+                     "COLL_NAME = '{{c}}' and DATA_NAME = '{{d}}'".format(c='{test_coll}',d='{filename}'),
+                     case_sensitive=True)
+                q2 = Query(callback,'DATA_ID',
+                     "COLL_NAME = '{{c}}' and DATA_NAME = '{{d}}'".format(c='{test_coll}'.upper(),d='{filename}'.lower()),
+                     case_sensitive=False)
+                L= [ len([i for i in q]) for q in (q1,q2) ]
+                callback.writeLine('stdout', repr(L).replace(chr(0x20),''))
+            OUTPUT ruleExecOut
+            ''')
+            print(rule_text.format(**locals()), file=f, end='')
+
+        output, err, rc = self.admin.run_icommand("irule -F " + rule_file)
+        self.assertTrue(rc == 0, "icommand status ret = {r} output = '{o}' err='{e}'".format(r=rc,o=output,e=err))
+        self.assertEqual(output.strip(), "[1,3]")
+
+    @unittest.skipIf(plugin_name == 'irods_rule_engine_plugin-irods_rule_language', 'only applicable for python REP')
+    def test_query_objects(self):
+        if not self.full_test : return
+        os.mkdir(self.dir_for_coll)
+        interesting = [0,1,2,100,254,255,256,257,258,510,511,512,513,514]
+        octals = ['%04o'%x for x in interesting]
+        max_count = sorted(interesting)[-1]
+        max_octal = sorted(octals)[-1]
+        for x in range(max_count+1):
+             fname = os.path.join(self.dir_for_coll,'%04o'%x)
+             with open(fname,'wb') as fb:
+                 fb.write(os.urandom(x))
+        lib.execute_command ('tar -cf {} {}'.format(self.bundled_coll, self.dir_for_coll))
+        self.admin.assert_icommand('icd')
+        self.admin.assert_icommand('iput -f {}'.format(self.bundled_coll))
+        self.admin.assert_icommand('ibun -x {} .'.format(self.bundled_coll))
+        test_coll = self.test_admin_coll_path
+        rule_file = ""
+        output=""
+        rule_header = dedent("""\
+            from genquery import *
+            def main(rule_args,callback,rei):
+                TestCollection = "{test_coll}"
+            """)
+        rule_footer = dedent("""\
+            INPUT null
+            OUTPUT ruleExecOut
+            """)
+
+        frame_rule = lambda text,indent=4: \
+            rule_header+('\n'+' '*indent+('\n'+' '*indent).join(dedent(text).split("\n"))
+                        )+"\n"+rule_footer
+        #--------------------------------
+        test_cases = [(
+            frame_rule('''\
+                L = []
+                for j in {octals}:
+                    cond_string= "COLL_NAME = '{{c}}' and DATA_NAME < '{{ltparam}}'".format(c=TestCollection,ltparam=j)
+                    q = Query(callback,
+                              columns=["DATA_NAME"],
+                              conditions=cond_string)
+                    L.append(q.total_rows())
+                callback.writeLine("stdout", repr(L))
+                '''),
+            "total_rows_",
+            lambda : eval(output) == interesting
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of possibly insecure function - consider using safer ast.literal_eval.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5016322)